test/integration/toolchains/gnu/test_autotoolsdeps.py

```diff
@@ -110,8 +109,8 @@ def generate(self):
                                       "-ldep2_onesystemlib -ldep2_twosystemlib "\
                                       "-ldep1_onesystemlib -ldep1_twosystemlib"
 
-                assert env["LDFLAGS"].startswith('dep1_shared_link_flag dep2_exe_link_flag dep1_exe_link_flag -framework dep2_oneframework -framework dep2_twoframework ' \
-                                     '-framework dep1_oneframework -framework dep1_twoframework ')
+                assert 'dep1_shared_link_flag dep2_exe_link_flag dep1_exe_link_flag -framework dep2_oneframework -framework dep2_twoframework ' \
+                                     '-framework dep1_oneframework -framework dep1_twoframework ' in env["LDFLAGS"]
```

Comment:
I'd like to understand better why this was failing and in which cases. The ``AutotoolsDeps`` ``env`` is supposed to be isolated.

Reply:
> I'd like to understand better why this was failing and in which cases. The ``AutotoolsDeps`` ``env`` is supposed to be isolated.

I found this test failing because I had defined `LDFLAGS` on my environment and conan tests does not provide a clean environment, it inherits the user shell environment. 

Changing `startswith` to `in` will allow the test to work even if the environment variable was already defined.

Reply:
I see. This might be made completely independent with ``env.unset("LDFLAGS")`` in the testing recipe as the first step to isolate from system env-vars.